spec/services/atmosphere/optimizer_spec.rb

```diff
@@ -43,25 +44,26 @@
         fl_32b.set_hourly_cost_for(Atmosphere::OSFamily.first, 10)
         fl_64b.set_hourly_cost_for(Atmosphere::OSFamily.first, 20)
 
-        selected_tmpl, selected_flavor = subject.select_tmpl_and_flavor([tmpl_64b])
+        selected_tmpl, selected_tenant, selected_flavor = subject.select_tmpl_and_flavor_and_tenant([tmpl_64b])
         expect(selected_tmpl).to eq tmpl_64b
+        expect(selected_tenant).to eq t
         expect(selected_flavor).to eq fl_64b
       end
     end
 
-    context 'is selected optimaly' do
+    context 'is selected optimally' do
       context 'appliance type preferences not specified' do
         it 'selects instance with at least 1.5GB RAM for public tenant' do
           appl_type = build(:appliance_type)
           tmpl = build(:virtual_machine_template, tenants: [amazon], appliance_type: appl_type)
-          selected_tmpl, flavor = subject.select_tmpl_and_flavor([tmpl])
+          selected_tmpl, tenant, flavor = subject.select_tmpl_and_flavor_and_tenant([tmpl])
```

Comment:
Useless assignment to variable - `selected_tmpl`. Use `_` or `_selected_tmpl` as a variable name to indicate that it won't be used.<br>Useless assignment to variable - `tenant`. Use `_` or `_tenant` as a variable name to indicate that it won't be used.<br>Line is too long. [91/80]